_posts/2020-06-01-spring-boot-hazelcast.md

```diff
@@ -0,0 +1,232 @@
+a cluster member. It happens by calling the method `Hazelcast.newHazelcastInstance()`. The method `getMap()`
+creates a `Map` in the cache or returns an existing one. The only thing we have to do to set the name of the `Map`.
+When we want to scale our application, every new instance will create a new member and this member will join
+the cluster automatically.
+
+This approach has two advantages.
+ * It is very easy to set up the cluster,
+ * Data access is very fast.
+ 
+We don't need to set up a separate cache cluster. It means we can create a cluster very fast by adding a couple
+lines of code.
+
+If we want to read the data from the cluster, the data access is low-latency, because we don't need to send
+a request to the cache cluster over the net.
+
+But it brings drawbacks too. Imagine we have a system, that requires one hundred instances of our application.
+In this cluster topology, it means we would have one hundred of cluster members even though we don't need it.
+The replication and synchronizing would be pretty expensive.
+Also, we have to note, that the `Hazelcast` is a java library. That means, the member can be embedded in a java
+application only.
+
+**Embedded cache topology should be used, when we have to execute high performing computing with the data from the cache.**
+
+### Configuration of Cluster
+Since we create the members of the cluster in our code, we can configure the cache.
+Let's look at a couple of the configuration parameters.
+
+````java
+@Component
+public class CacheClient {
+
+    public static final String CARS = "cars";
+    private final HazelcastInstance hazelcastInstance 
+         = Hazelcast.newHazelcastInstance(createConfig());
+
+    public Config createConfig() {
+        Config config = new Config();
+        config.addMapConfig(mapConfig());
+        return config;
+    }
+
+    private MapConfig mapConfig() {
+        MapConfig mapConfig = new MapConfig(CARS);
+        mapConfig.setTimeToLiveSeconds(360);
+        mapConfig.setMaxIdleSeconds(20);
+        return mapConfig;
+    }
+    
+    // Other methods omitted.
+}
+````
+
+We can configure every `Map` or other data structure in the cluster separately. In this case, we
+configure the `Map` Cars. 
+
+With the method `setTimeToLiveSeconds(360)` we define how long an
+entry stays in the cache. After 360 seconds, the entry will be evicted. If the entry is updated, the eviction time is reset to 0 again.
+
+The method `setMaxIdleSeconds(20)` defines how long the entry stays in the cache without being touched.
+Entry is touched by any reading operation. If an entry is not touched for 20 seconds, it will be evicted.
+ 
+
+## Client-Server Topology
+This type of topology means, that we set up a separate cache cluster, and our application is a client of this
+cluster.
+
+![Client-Server Cache Topology](/assets/img/posts/hazelcast/client-server-cache.png)
```

Comment:
Did you make these diagrams yourself? Or did you get them from somewhere? Just asking because of potential licensing stuff.

Reply:
I drew the diagrams myself.